packages/strapi-plugin-upload/controllers/Upload.js

```diff
@@ -41,6 +41,7 @@ module.exports = {
       // Add details to the file to be able to create the relationships.
       if (refId && ref && field) {
         Object.assign(file, {
+          path: `${ref}/${field}`,
```

Comment:
When uploading through the S3 service, this part is unreachable because `(refId && ref && field)` all three are undefined, so `path` is never set.

Reply:
yes, you are right, if the upload file does not have relationship, it will cause undefined folder problem.